syntax_and_semantics/capturing_blocks.md

```diff
@@ -2,7 +2,9 @@
 
 A block can be captured and turned into a `Proc`, which represents a block of code with an associated context: the closured data.
 
-To capture a block you must specify it as a method's block argument, give it a name and specify the input and output types. For example:
+To capture a block you must specify it as a method's block argument, and return that block from the method.
```

Comment:
A method doesn't have to return a block to capture it. It can be captured even if it is not used outside the method.
You should keep the mention of input and output types.

Reply:
@straight-shoota - Understood about the "return that block" statement being incorrect.
The statement "specify the input and output types" - that reads to me that you can't do any of these:
`Int32 ->` - no output
`-> String` - no input
`Int32 -> _` - generic output

It may need to stay, but I think it needs a rewording, what do you think of:
> ... specify the inputs and outputs. You can also specify that there are no inputs and/or there is no output. Input types must be specific, the output can be specific or generic.

Reply:
@straight-shoota - Would this be a true statement?
> To capture a block you must specify it as a method's block argument, then store it in a variable or return it from the method.

If it isn't stored in a variable (instance, class, or global) and it isn't returned from the method - is there any other way to "capture" it?

Reply:
Hm. Specifying input and output types IMHO also includes specifying none.
But I understand that the original wording might be confusing. Maybe it's better to use `type declaration` instead of `type`? Making it clear that it doesn't have to be a literal type but you can also declare that there is none. For example with your last suggestion: `... specify it as a method's block argument including input and output type declarations`.

I would leave out the part `or return it from the method` because to be able to return it, you need to have it in a variable (including local) and therefore it get's captured.

In this example, the block is only inlined. Still, it technically is captured in the variable `block` and does qualify as a captured block, though it is not *used* as such:
```
def capture(&block : -> Int32)
  yield
end
```

Reply:
No, that's not a captured block. A captured block is only one where it's mentioned in the body of the method.